### PR TITLE
Clone

### PR DESCRIPTION
**Pull request by ioedeveloper**
Closes https://github.com/ethereum/remix-project/issues/2451

### CODE REVIEW

File: libs/remix-ui/settings/src/lib/github-settings.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    setGithubEmail('')
+    props.removeTokenToast()
+  }
+  
+  return (
+    <div className="border-top">
+      <div className="card-body pt-3 pb-2">
+        <h6 className="card-title">GitHub Access Token</h6>
+        <p className="mb-1">Manage the access token used to publish to Gist and retrieve GitHub contents.</p>
+        <p className="">Go to github token page (link below) to create a new token and save it in Remix. Make sure this token has only \'create gist\' permission.</p>
```
Comment (Aniket-Engg) [inline libs/remix-ui/settings/src/lib/github-settings.tsx:51]: If this token will also be used for repo cloning, it should be mentioned here.

File: libs/remix-ui/settings/src/lib/github-settings.tsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+          </div>
+        </div>
+        <div>
+          <label>USERNAME:</label>
+          <div className="text-secondary mb-0 h6">
+            <input id="githubusername" data-id="settingsTabGithubUsername" type="text" className="form-control" onChange={(e) => handleChangeUserNameState(e)} value={ githubUserName } />
+          </div>
+        </div>
+        <div>
+          <label>EMAIL:</label>
```
Comment (Aniket-Engg) [inline libs/remix-ui/settings/src/lib/github-settings.tsx:69]: Do we require username and email both for repo cloning?

File: libs/remix-ui/workspace/src/lib/components/clone.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import React from 'react'
+import { OverlayTrigger, Popover } from 'react-bootstrap';
+
+const popover = (
+  <Popover id="popover-basic" className='bg-light border-secondary'>
+    <Popover.Title as="h3" className='bg-dark border-0'>Clone Repository</Popover.Title>
```
Comment (Aniket-Engg) [inline libs/remix-ui/workspace/src/lib/components/clone.tsx:6]: Title can include `GitHub` as `Clone GitHub Repository`

File: libs/remix-ui/workspace/src/lib/components/clone.tsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+      <Popover.Content>
+        <div className="remixui_cloneContainer">
+          <input
+            className="form-control"
+            placeholder="Enter github repository url"
```
Comment (Aniket-Engg) [inline libs/remix-ui/workspace/src/lib/components/clone.tsx:11]: github can be `GitHub`

File: libs/remix-ui/workspace/src/lib/components/clone.tsx [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+            title="Enter github repository url"
+          />
+        </div>
+      </Popover.Content>
+  </Popover>
+)
+
+export const CloneRepository = () => (
+  <OverlayTrigger trigger="click" placement={"bottom"} overlay={popover}>
+    <i className="fas fa-cloud-download remixui_menuicon pr-0 mr-2"></i>
```
Comment (Aniket-Engg) [inline libs/remix-ui/workspace/src/lib/components/clone.tsx:21]: `title` can also be added

**Comment by Aniket-Engg** [comment]
I am not sure if UI for this is finalized but I find Modal for this more user friendly.

**Comment by Aniket-Engg** [comment]
Other observations:

- If I click on cloud icon, it shows a popover, still I can interact with outer things and it is not disappeared on any click until I again click on cloud icon. Not sure if it is a design choice
Screenshot 2022-06-22 at 2 15 05 PM

- After putting the url in the textbox, it seems too small to me
Screenshot 2022-06-22 at 2 16 40 PM

- I was not sure what to do after putting the URL. I just clicked ENTER and no message appeared neither it did cloning

File: nx.json [hunk 6]
```diff
@@ -169,14 +169,9 @@
     },
     "remix-ui-permission-handler": {
       "tags": []
+    },
+    "remix-ui-tooltip-popup": {
+      "tags": []
```
Comment (bunsenstraat) [inline nx.json:174]: don't remove the deps build

**Review by Aniket-Engg — COMMENTED** [review]
Even after setting github details, I couldn't make cloning work. See other comments.

**Review by ryestew — COMMENTED** [review]
The "clone" workflow from the FE (that this PR brings) is simpler than using DGIT. 

I assume that it only brings in 1 level and only brings in the main/master branch.   Is this the case?

**Comment by ioedeveloper** [comment]
> The "clone" workflow from the FE (that this PR brings) is simpler than using DGIT.
> 
> I assume that it only brings in 1 level and only brings in the main/master branch. Is this the case?

Yes, that is the case for now.

**Review by ryestew — APPROVED** [review]

**Pull request merged**